src/Trig.cpp

```diff
@@ -51,6 +51,17 @@ float NAS2D::angleFromPoints(float x, float y, float x2, float y2)
 }
 
 
+/**
+ * Gets a directional vector from an angle in degrees.
+ */
+Vector<float> NAS2D::angleToDirection(float angle)
+{
+	// static_cast<float> used to suppress warning at possible loss of data. Intentionally
+	// surpressed as we don't need that level of precision.
+	return {static_cast<float>(std::sin(NAS2D::degToRad(angle))), static_cast<float>(-std::cos(NAS2D::degToRad(angle)))};
```

Comment:
This is incorrect. The formula is `x = std::cos(degToRad(angle)); y = std::sin(degToRad(angle))`

The cast to float is not needed either, there are overloads that return a `float`.

Reply:
Oh yeah, the `float` overloads were added in C++11. For some reason I thought that was an MSVC thing. I did actually try removing the casts locally and saw there was no conversion warning.

Reference: [`std::sin`](https://en.cppreference.com/w/cpp/numeric/math/sin)

----

I'll have to give the rotation formula some consideration. It will depend on things like orientation of the coordinate system, and orientation of the angle. Perhaps we should open an Issue and change that as a separate PR.

Reply:
Hmm, seems there is similar `static_cast` for other trig functions too, such as [`std::atan2`](https://en.cppreference.com/w/cpp/numeric/math/atan2).

Reply:
I've opened issue #519 to track the formula used to convert from an `angle` to a `Vector`.

Reply:
The above referenced Rotation Matrix only applies if you are rotating a point in space. `angleToDirection` isn't doing that. It's calculating a unit vector given an angle:
https://en.wikipedia.org/wiki/Trigonometric_functions#Unit-circle_definitions

>The trigonometric functions cos and sin are defined, respectively, as the x- and y-coordinate values of point A, i.e., cos [theta] = Ax and sin [theta] = Ay

Reply:
Additionally, this function should really be moved into `Vector` itself as it has no equivalent for three- and four-dimensional vectors.

Reply:
The exact formula used is perhaps out of scope for this PR, and so better addressed as a separate issue. This change is more concerned with the return type.

----

Adding the method to `Vector` is an interesting idea. Though in this case it's constructing a `Vector` from a scalar value. That would imply a factory function, rather than a member function.

Reply:
I was thinking, the reverse direction, which calculates the angle from a `Vector` would make for a good member function.

Probably best to name it something like `angle`, rather than the more mathematical name `arg` (argument). Though there may need to be a distinction for radians versus degrees. Or we could just default to always using radians.

----

If I add an `angle` method, I would probably add the reverse factory method at the same time.

----

I would probably want to use strict mathematical conventions for something implemented directly in `Vector`.

For other conventions, such as screen coordinates, we'd probably want to keep those helper methods outside of `Vector`.

Reply:
https://github.com/cugone/Abrams2019/blob/master/Engine/Code/Engine/Math/Vector2.hpp

- `CalcHeadingRadians/Degrees` for the getting the angle of a vector (`atan2(y/x)`)
- `SetHeadingRadians/Degrees` for setting the angle via `R * cos and sin` without changing the length.
- `SetUnitLengthAndHeadingRadians/Degrees` for setting x/y directly with `cos(angle)`/`sin(angle)`

Reply:
Just had a thought about moving this stuff over to `Vector`. The result `angle` will always be a floating point value, even if `BaseType` is `int`. Similarly, you might have a `Vector<double>`. Should that return a `float` angle, or a `double` angle?

Should it be possible to construct a `Vector<int>` from an angle?

I'm kind of leaning away from including this directly in `Vector` for the moment. It seems these methods are more useful for `Vector<float>`, rather than generic `Vector` types.

Reply:
> 
> 
> Just had a thought about moving this stuff over to `Vector`. The result `angle` will always be a floating point value, even if `BaseType` is `int`. Similarly, you might have a `Vector<double>`. Should that return a `float` angle, or a `double` angle?
> 
> Should it be possible to construct a `Vector<int>` from an angle?
> 
> I'm kind of leaning away from including this directly in `Vector` for the moment. It seems these methods are more useful for `Vector<float>`, rather than generic `Vector` types.

This is where trying to be generic all the time is going to hurt and make life more complicated for no reason. It'd be much simpler to just have `Vector2/3/4` and `IntVector2/3/4`. `double` and `long double`s are abhorred by both GPUs and the gaming industry. They just don't use them due to space, alignment, and padding requirements.

Reply:
I suppose if you wanted to get specific with types, you could create a `VectorFloat` class that derived from `Vector<float>` and implemented extension methods.

I'm not sure it's really worth it though.

I'm tempted to leave the conversion as free standing methods.